libcloudforensics/providers/azure/internal/account.py

```diff
@@ -381,6 +387,314 @@ def CreateDiskFromSnapshotURI(
+
+    Args:
+      region (str): Optional. The region in which to look the instance types.
+          By default, look in the default_region associated to the AZAccount
+          object.
+
+    Returns:
+      List[Dict[str, str]]: A list of available vm size. Each size is a
+          dictionary containing the name of the configuration, the number of
+          CPU cores, and the amount of available memory (in MB).
+          E.g.: {'Name': 'Standard_B1ls', 'CPU': 1, 'Memory': 512}
+    """
+    if not region:
+      region = self.default_region
+    available_vms = self.compute_client.virtual_machine_sizes.list(region)
+    vm_sizes = []
+    for vm in available_vms:
+      vm_sizes.append({
+          'Name': vm.name,
+          'CPU': vm.number_of_cores,
+          'Memory': vm.memory_in_mb
+      })
+    return vm_sizes
+
+  def _GetInstanceType(self, cpu_cores: int, memory_in_mb: int) -> str:
+    """Returns an instance type for the given number of CPU cores / memory.
+
+    Args:
+      cpu_cores (int): The number of CPU cores.
+      memory_in_mb (int): The amount of memory (in MB).
+
+    Returns:
+      str: The instance type for the given configuration.
+
+    Raises:
+      ValueError: If no instance type matches the requested configuration.
+    """
+    vm_sizes = self.ListInstanceTypes()
+    for size in vm_sizes:
+      if size['CPU'] == cpu_cores and size['Memory'] == memory_in_mb:
+        instance_type = size['Name']  # type: str
+        return instance_type
+    raise ValueError(
+        'No instance type found for the requested configuration: {0:d} CPU '
+        'cores, {1:d} MB memory.'.format(cpu_cores, memory_in_mb))
+
+  def _CreateNetworkInterfaceForVM(self,
+                                   vm_name: str,
+                                   region: Optional[str] = None) -> str:
+    """Create a network interface and returns its ID.
+    This is necessary when creating a VM from the SDK.
+    See https://docs.microsoft.com/en-us/azure/virtual-machines/windows/python
+
+    Args:
+      vm_name (str): The name of the VM to create the network interface for.
+      region (str): Optional. The region in which to create the network
+          interface. Default uses default_region of the AZAccount object.
+
+    Returns:
+      str: The id of the created network interface.
+
+    Raises:
+      ValueError: if vm_name is not provided.
+      RuntimeError: If no network interface could be created.
+    """
+    if not vm_name:
+      raise ValueError('vm_name cannot be None.')
+
+    if not region:
+      region = self.default_region
+
+    network_interface_name = '{0:s}-nic'.format(vm_name)
+    ip_config_name = '{0:s}-ipconfig'.format(vm_name)
+
+    # Check if the network interface already exists, and returns its ID if so.
+    try:
+      nic = self.network_client.network_interfaces.get(
+          self.default_resource_group_name, network_interface_name)
+      nic_id = nic.id  # type: str
+      return nic_id
```

Comment:
Maybe we can `return nic.id` directly?

Reply:
This would break mypy:

```
libcloudforensics/providers/azure/internal/account.py:601: error: Returning Any from function declared to return "str"
```

Not that you cannot specify the type on the return line either:
```
libcloudforensics/providers/azure/internal/account.py:601: error: misplaced type annotation
```